sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/ModelsRepositoryAsyncClient.java

```diff
@@ -62,4 +73,71 @@
     public ModelsRepositoryServiceVersion getServiceVersion() {
         return this.serviceVersion;
     }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmi A well-formed DTDL model Id. For example 'dtmi:com:example:Thermostat;1'.
+     * @return A Map containing the model definition(s) where the key is the dtmi
+     * and the value is the raw model definition string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Map<String, String>> GetModels(String dtmi) {
+        return GetModels(dtmi, this.defaultDependencyResolutionOption);
+    }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmi                       A well-formed DTDL model Id. For example 'dtmi:com:example:Thermostat;1'.
+     * @param dependencyResolutionOption A DependencyResolutionOption value to force model resolution behavior.
+     * @return A Map containing the model definition(s) where the key is the dtmi
+     * and the value is the raw model definition string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Map<String, String>> GetModels(String dtmi, DependencyResolutionOptions dependencyResolutionOption) {
+        return withContext(context -> {
+            try {
+                return GetModels(dtmi, dependencyResolutionOption, context);
+            } catch (Exception e) {
+                e.printStackTrace();
+                return null;
+            }
+        });
+    }
+
+    private Mono<Map<String, String>> GetModels(String dtmi, DependencyResolutionOptions dependencyResolutionOption, Context context) throws Exception {
+        context.addData(AZ_TRACING_NAMESPACE_KEY, MODELS_REPOSITORY_TRACING_NAMESPACE_VALUE);
+        return repositoryHandler.ProcessAsync(dtmi, dependencyResolutionOption, context);
+    }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmis Collection of well-formed DTDL model Ids
+     * @return A Map containing the model definition(s) where the key is the dtmi
+     * and the value is the raw model definition string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Map<String, String>> GetModels(Iterable<String> dtmis) {
+        return GetModels(dtmis, this.defaultDependencyResolutionOption);
+    }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmis                      Collection of well-formed DTDL model Ids.
```

Comment:
"**Iterable** of well-formed DTDL model Ids" since this is Java, not .NET

Reply:
a Collection is a less generic class, and we don't need it to be a Collection, so the Iterable input variable is correct here